stable/fairwinds-insights/templates/postgresql-cluster.yaml

```diff
@@ -0,0 +1,276 @@
+            
+            if [ -n "$EXISTING_OPERATOR" ]; then
+              echo "Using existing CloudNativePG operator in namespace: $EXISTING_OPERATOR"
+              OPERATOR_NAMESPACE="$EXISTING_OPERATOR"
+            else
+              echo "Installing CloudNativePG operator in namespace: $OPERATOR_NAMESPACE"
+                
+                # Create namespace if it doesn't exist
+                if ! kubectl get namespace "$OPERATOR_NAMESPACE" 2>/dev/null; then
+                  kubectl create namespace "$OPERATOR_NAMESPACE"
+                fi
+                
+                # Install CloudNativePG operator using standard YAML manifest
+                echo "Installing CloudNativePG operator using standard YAML manifest..."
+                kubectl apply --server-side -f https://raw.githubusercontent.com/cloudnative-pg/cloudnative-pg/release-1.27/releases/cnpg-1.27.0.yaml || true
+                
+                # Verify the installation as per CloudNativePG documentation
+                echo "Verifying CloudNativePG operator installation..."
+                kubectl get deploy -n cnpg-system cnpg-controller-manager
+                
+                # Wait for resources to be created
+                sleep 10
+              
+              # Check if the operator deployment exists
+              for i in {1..5}; do
+                if kubectl get deployment -n cnpg-system cnpg-controller-manager >/dev/null 2>&1; then
+                  echo "CloudNativePG operator installed successfully!"
+                  OPERATOR_NAMESPACE="cnpg-system"
+                  break
+                else
+                  sleep 5
+                fi
+              done
+              
+              # Final check
+              if ! kubectl get deployment -n cnpg-system cnpg-controller-manager >/dev/null 2>&1; then
+                echo "ERROR: Failed to install CloudNativePG operator"
+                exit 1
+              fi
+            fi
+            
+            # Wait for webhook service to be ready
+            echo "Waiting for webhook service to be ready..."
+            kubectl wait --for=condition=available --timeout=120s deployment/cnpg-webhook-server -n cnpg-system || true
+            
+            # Create the PostgreSQL cluster
+            echo "Creating PostgreSQL cluster..."
+            cat <<EOF | kubectl apply -f -
+            apiVersion: postgresql.cnpg.io/v1
+            kind: Cluster
+            metadata:
+              name: insights-postgres
+              namespace: {{ .Release.Namespace }}
+              labels:
+                {{- include "fairwinds-insights.labels" . | nindent 16 }}
+            spec:
+              instances: 1
+              enableSuperuserAccess: true
+              imageName: {{ .Values.postgresql.image.registry }}/{{ .Values.postgresql.image.repository }}:{{ .Values.postgresql.image.tag }}
+              postgresql:
+                parameters:
```

Comment:
can we make `parameters` a configurable object?

Reply:
done